src/main/java/minijava/parser/Parser.java

```diff
@@ -153,8 +160,9 @@ private void parseMethod() {
   /** Parameters -> Parameter | Parameter , Parameters */
   private void parseParameters() {
     parseParameter();
-    if (isCurrentTokenTypeOf(COMMA)) {
-      parseParameters();
+    while (isCurrentTokenTypeOf(COMMA)) {
+      expectAndConsume(COMMA);
+      parseParameter();
```

Comment:
Also I got rid of the tail call, which, as we experienced, the JVM handles quite poorly...